main.go

```diff
@@ -15,8 +15,13 @@ func main() {
 	flag.Parse()
 
 	addr := ":" + *port
-	handler := http.HandlerFunc(handle)
+	server := server{}
+	handler := http.HandlerFunc(server.handleMain)
 
-	fmt.Printf("http://localhost:%s\n", *port)
-	log.Fatal(http.ListenAndServe(addr, handler))
+	fmt.Printf("http://localhost%s\n", addr)
+	go func() {
+		log.Fatal(http.ListenAndServe(addr, handler))
+	}()
+
+	log.Fatal(server.listenInput())
```

Comment:
I'd actually invert these, `listenInput()` would go in a goroutine because it's not the _main_ process to run.
`http.ListenAndServe()` would still do its forever loop as it always does.